kernel/comps/logger/src/default_logger.rs

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use log::{Metadata, Record};
+use ostd::timer::Jiffies;
+use spin::Once;
+
+/// Default logger of this crate.
+///
+/// This logger only has the most basic logging functionality, controls the output
+/// based on the globally set log level, and different log levels are represented
+/// with different colors.
+pub(super) struct DefaultLogger {}
+
+pub(super) static LOGGER: Once<DefaultLogger> = Once::new();
+
+impl log::Log for DefaultLogger {
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        metadata.level() <= log::max_level()
+    }
+
+    fn log(&self, record: &Record) {
+        let timestamp = Jiffies::elapsed().as_duration().as_secs_f64();
+        let level = record.level();
+
+        // Use a global lock to prevent interleaving of log messages.
+        use ostd::sync::SpinLock;
+        static RECORD_LOCK: SpinLock<()> = SpinLock::new(());
+        let _lock = RECORD_LOCK.disable_irq().lock();
+
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "log_color")]{
```

Comment:
*This is actually a comment on the original logger implementation. I admit that the problem is not introduced by this PR, but it is just slightly better if it can be addressed together.*

The way the `log_color` function is implemented is too ugly. Using macros creates unnecessary nested indentation and makes the code formatter not work (as you can see below, `}else{` isn't formatted properly).

I suggest to do this:
```rust
#[cfg(feature = "log_color")]
fn print_logs() { /* ... */ }

#[not(cfg(feature = "log_color"))]
fn print_logs() { /* ... */ }
```
Then just call `print_logs` here.